extensions/wikia/FeedsAndPosts/JustUpdatedAPIProxy.class.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Wikia\FeedsAndPosts;
+
+class JustUpdatedAPIProxy {
+
+	const LIMIT = 4;
+
+	private function requestAPI($params) {
+		$api = new \ApiMain(new \FauxRequest($params));
+		$api->execute();
+
+		return $api->GetResultData();
+	}
+
+	private function getRecentChanges() {
+		$response = $this->requestAPI( [
+			'action' => 'query',
+			'list' => 'recentchanges',
+			'rcprop' => 'title',
+			'rcshow' => '!bot|!minor|!redirect',
+			'rcnamespace' => '0',
+			'rclimit' => self::LIMIT,
+		] );
+
+		if ( isset( $response['query']['recentchanges'] ) ) {
+			return $response['query']['recentchanges'];
+		}
+
+		return [];
+	}
+
+	private function getImage($title) {
+		$response = $this->requestAPI( [
+			'action' => 'imageserving',
+			'wisTitle' => $title,
+		] );
+
+		if ( isset( $response['image']['imageserving'] ) ) {
+			return $response['image']['imageserving'];
+		}
+
+		return null;
+	}
+
+	public function get() {
+		return \WikiaDataAccess::cacheWithOptions(
```

Comment:
instead of using `cacheWithOptions` you can use `WikiaDataAccess::cache(wfMemcKey( 'feeds-just-updated' ), WikiaDataAccess::CACHE_VERY_SHORT, function(){...});` the command is `USE_CACHE` by default